fix: Image files are saved as different file names when using drag-and-drop (3119)

kiyoka opened the pull request:
<!-- Please change the Answers in the table below
     to reflect the contents of your pull request. -->

| Q                 | A
| ----------------- | ---
| Bug fix?          | yes
| New feature?      | no
| Breaking changes? | no
| Deprecations?     | no
| New tests added?  | not needed
| Fixed tickets     | Fixes #3119
| License           | MIT

### Description
New files are not created for files that already exist in the image assets folder.

kiyoka committed:
fix: #3119 Image files are saved as different file names when using drag-and-drop

- Added a function IsIdenticalImagePath()
- don't save new file when image file is exists in assets folder.

fxha commented on src/renderer/util/fileSystem.js:
```diff
@@ -33,6 +33,44 @@ export const getContentHash = content => {
+ * @param {String} srcImagePath The source image file path.
+ * @returns {{identical: boolean, relPath: string}} Returns the idencial or not and relativePath of image file.
+ */
+export const IsIdenticalImagePath = async (cwd, relativeName, srcImagePath) => {
```
`async` isn't needed because no `await` inside.

fxha replied:
Please use lower-case `if` and could we find a better name as the logic is to return the relative path if identical?

fxha replied:
I think you can use `isChildOfDirectory` from `common/filesystem` to keep code clean? The idea is to check whether the source image is inside the assets directory right?

kiyoka committed:
Merge branch 'develop' into fix-issue-3119

kiyoka committed:
Applying Review Findings

- Changed logic to check if the image path is an assets path.
- Renamed funciton.
- Removed 'async' and 'await' keyword.

kiyoka commented:
@fxha 
Review results applied.
Please review again.

Jocs commented:
This project recently merged a large refactor (electron-vite migration, [PR #4001](https://github.com/marktext/marktext/pull/4001)) that upgraded dependencies and fixed several issues.

Please review this PR and let us know if it is still needed, or if the changes have already been addressed upstream.

> **Note:** This PR currently has merge conflicts with the `develop` branch. If you decide to continue with this PR, please rebase or merge the latest `develop` branch to resolve the conflicts before it can be reviewed.

Jocs commented:
Thank you for the contribution. Since this PR was opened, the project has gone through a significant refactor to electron-vite (#4001) — `src/renderer/util/fileSystem.js` is now `src/renderer/src/util/fileSystem.js`, and `moveImageToFolder()` has been reworked. The current diff cannot be applied to `develop`.

Closing this PR so it doesn't get stuck in limbo. If the drag-and-drop image-naming issue (#3119) still reproduces on the current `develop` branch, please feel free to open a new PR on top of the latest code — we'd be happy to review it.
